File: WoofAndroid/data/datasource/src/main/kotlin/com/perrystreet/woof/datasource/account/AccountLocalDataSource.kt
package com.perrystreet.woof.datasource.account

import com.perrystreet.woof.datasource.dogs.fixtures.DogFixtures
import com.perrystreet.woof.dto.dog.DogProfileDTO
import com.perrystreet.woof.utils.scheduler.ISchedulerProvider
import io.reactivex.rxjava3.core.Single
import java.util.concurrent.TimeUnit

@org.koin.core.annotation.Single
class AccountLocalDataSource(
    private val scheduler: ISchedulerProvider,
) : IAccountDataSource {

    override fun getAccount(): Single<DogProfileDTO> =
        Single.fromCallable { DogFixtures.account() }
            .delay(LatencyMillis, TimeUnit.MILLISECONDS, scheduler.computationScheduler)
            .observeOn(scheduler.mainScheduler)

    private companion object {
        const val LatencyMillis = 600L
    }
}

File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSource/Account/AccountLocalDataSource.swift
import Combine
import DI
import DTO
import Foundation
import Utils

@SingleForProtocol
public final class AccountLocalDataSource: AccountDataSourceImplementing {
    private let scheduler: SchedulerProviding

    private static let latencyMillis = 600

    public func getAccount() -> AnyPublisher<DogProfileDTO, DataSourceError> {
        Just(DogFixtures.account())
            .setFailureType(to: DataSourceError.self)
            .delay(for: .milliseconds(Self.latencyMillis), scheduler: scheduler.computationScheduler)
            .receive(on: scheduler.mainScheduler)
            .eraseToAnyPublisher()
    }
}
